logs/querysample/querysample.go

```diff
@@ -31,6 +32,21 @@ func TransformAutoExplainToQuerySample(logLine state.LogLine, explainText string
 func transformExplainJSONToQuerySample(logLine state.LogLine, explainText string, queryRuntimeMs float64) (state.PostgresQuerySample, error) {
 	var explainJSONOutput state.ExplainPlanContainer
 
+	// With Heroku + JSON format, the value of "Query Text" can contain \n (new line).
+	// JSON's value can't contain \n, so replacing it with a white space
+	if cUtil.IsHeroku() {
+		re := regexp.MustCompile(`(?s)("Query Text"\s*:\s*")(.*?)"`)
+		explainText = re.ReplaceAllStringFunc(explainText, func(m string) string {
+			// Extract inner value and flatten it
+			submatches := re.FindStringSubmatch(m)
+			key := submatches[1]
+			val := submatches[2]
+			// Remove newlines and replace with space
+			cleaned := regexp.MustCompile(`[\r\n]+`).ReplaceAllString(val, " ")
+			return key + cleaned + `"`
+		})
+	}
+
 	if err := json.Unmarshal([]byte(explainText), &explainJSONOutput); err != nil {
```

Comment:
As discussed in PR review, it may be useful to see if we can find a JSON parser that is more lenient with newlines, as an alternate to the regexp-based approach.

Reply:
I looked around this (more lenient one is called "json5"), but none of json5 modules of golang is popular enough or widely used.
According to JSON RFC ([RFC 8259](https://datatracker.ietf.org/doc/html/rfc8259)), the control characters including newlines must be escaped, and also golang community is not really keen to support json5 in general according to [the forum post](https://groups.google.com/g/golang-nuts/c/mhj1D8HKZNU).

With that, I'd say that let's start with this.